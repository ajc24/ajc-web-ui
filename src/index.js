/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
