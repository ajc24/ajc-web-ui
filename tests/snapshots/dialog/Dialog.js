import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { Dialog } from '../../../src';

describe('Dialog', () => {
  const twoButtonsList = [
    {
      id: 'save-button',
      onClick: () => {},
      title: 'Save',
      type: 'submit',
    },
    {
      id: 'cancel-button',
      onClick: () => {},
      title: 'Cancel',
      type: 'button',
    }
  ];

  describe('Default props and rendering - Component with grey theme, white background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <Dialog id="test-dialog-id" titleTextContent="Test Dialog Title">
            Test Dialog Content.
          </Dialog>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with red theme, grey background and buttons list', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <Dialog id="test-dialog-id" titleTextContent="Test Dialog Title" colour="red" bgColour="grey" buttonsList={twoButtonsList}>
            Test Dialog Content.
          </Dialog>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with grey theme, yellow background and buttons list', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <Dialog id="test-dialog-id" titleTextContent="Test Dialog Title" colour="grey" bgColour="yellow" buttonsList={twoButtonsList}>
            Test Dialog Content.
          </Dialog>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
