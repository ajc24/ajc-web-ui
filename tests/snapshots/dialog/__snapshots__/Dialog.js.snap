// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Default props and rendering - Component with grey theme, white background verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-width-dialog-parent"
    id="test-dialog-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-title ajc-width-dialog ajc-text-white ajc-font-default ajc-background-grey-2"
      id="test-dialog-id-title"
    >
      <h1
        className="ajc-font-dialog-title"
      >
        Test Dialog Title
      </h1>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-white"
      id="test-dialog-id-content-area-root"
    >
      <div
        className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
        id="test-dialog-id-content-area"
      >
        Test Dialog Content.
      </div>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-grey-2"
      id="test-dialog-id-footer"
    />
  </div>
</div>
`;

exports[`Dialog Transferred props and rendering - Component with grey theme, yellow background and buttons list verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-width-dialog-parent"
    id="test-dialog-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-title ajc-width-dialog ajc-text-white ajc-font-default ajc-background-grey-2"
      id="test-dialog-id-title"
    >
      <h1
        className="ajc-font-dialog-title"
      >
        Test Dialog Title
      </h1>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-yellow"
      id="test-dialog-id-content-area-root"
    >
      <div
        className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
        id="test-dialog-id-content-area"
      >
        Test Dialog Content.
      </div>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-grey-2"
      id="test-dialog-id-footer"
    >
      <button
        className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-grey"
        id="save-button"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
      <button
        className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-grey"
        id="cancel-button"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog Transferred props and rendering - Component with red theme, grey background and buttons list verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-width-dialog-parent"
    id="test-dialog-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-title ajc-width-dialog ajc-text-white ajc-font-default ajc-background-red"
      id="test-dialog-id-title"
    >
      <h1
        className="ajc-font-dialog-title"
      >
        Test Dialog Title
      </h1>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-grey-1"
      id="test-dialog-id-content-area-root"
    >
      <div
        className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
        id="test-dialog-id-content-area"
      >
        Test Dialog Content.
      </div>
    </div>
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-red"
      id="test-dialog-id-footer"
    >
      <button
        className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
        id="save-button"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
      <button
        className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
        id="cancel-button"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
